server/src/rbx_snapshot.rs

```diff
@@ -315,13 +336,13 @@ fn snapshot_imfs_directory<'source>(
         snapshot_imfs_path(context, imfs, &init_client_path, Some(snapshot_name))?.unwrap()
     } else {
         RbxSnapshotInstance {
-            class_name: Cow::Borrowed("Folder"),
+            class_name: Cow::Owned(meta.class_name.unwrap_or_else(|| "Folder".to_owned())),
             name: snapshot_name,
-            properties: HashMap::new(),
+            properties: meta.properties,
             children: Vec::new(),
             metadata: MetadataPerInstance {
                 source_path: None,
-                ignore_unknown_instances: false,
+                ignore_unknown_instances: meta.ignore_unknown_children.unwrap_or(false),
                 project_definition: None,
             },
         }
```

Comment:
Instead of applying the metadata in the else branch here, can we apply the metadata unconditionally after this branch? It might even be a good idea to defer loading the `init.meta.json` file until that point.

This branch is only applied when there is no `init.lua`, `init.server.lua`, or `init.client.lua` file in the folder. It's still useful to apply metadata to the instance even if it becomes a script by one of those mechanisms!

Reply:
I believe I did what you asked.